plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java

```diff
@@ -62,21 +86,77 @@ public void apply(BufferedImage image) {
         int width = image.getWidth();
         int height = image.getHeight();
 
+
+//        logger.info("width: " + width);
+//        logger.info("height: " + height);
+        /*
+            NOTE: different sizes:
+
+            108 x 108
+            162 x 162
+            432 x 432
+            324 x 324
+            216 x 216
+            48 x 48
+            72 x 72
+            192 x 192
+            144 x 144
+         */
+
         Graphics2D g = (Graphics2D) image.getGraphics();
 
-        g.setTransform(AffineTransform.getRotateInstance(Math.toRadians(-45)));
+        int transformAmount = 0;
 
-        int y = height / (largeRibbon ? 2 : 4);
+        switch (position) {
+            case TopHorizontal:
+            case BottomHorizontal:
+                transformAmount = 0;
+                break;
+            case Default:
+                transformAmount = -45;
+                break;
+        }
+
+        // transform
+        g.setTransform(AffineTransform.getRotateInstance(Math.toRadians(transformAmount)));
+
+        int y = height / 2;
+        int x = 0;
 
-        // calculate the rectangle where the label is rendered
         FontRenderContext frc = new FontRenderContext(g.getTransform(), true, true);
+        // calculate the rectangle where the label is rendered
         int maxLabelWidth = calculateMaxLabelWidth(y);
-        g.setFont(getFont(maxLabelWidth, frc));
+        g.setFont(getFont(height, maxLabelWidth, frc));
+
         Rectangle2D labelBounds = g.getFont().getStringBounds(label == null ? "" : label, frc);
+        int labelHeight = (int) labelBounds.getHeight();
+
+        int padding = largeRibbon ? (labelHeight / 8) : (labelHeight / 10);
```

Comment:
Tbh, this is probably a better way to handle adaptive icons than the current one 👍 . 
I just need to test different scenarios to see how it looks.